### PR TITLE
fix #1017 spilt project core

### LINKED ISSUE #1017: Split Gearpump-core into 2 sub projects.

### PR DESCRIPTION
**Pull request by huafengw**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -2,7 +2,7 @@ addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.2")
 
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.11.2")
 
-addSbtPlugin("org.xerial.sbt" % "sbt-pack" % "0.6.9")
+addSbtPlugin("org.xerial.sbt" % "sbt-pack" % "0.6.12")
```
Comment (clockfly) [inline project/plugins.sbt]: Why changing the version? Have you tested fully on linux, windows?

**Comment by clockfly** [comment]
+1

File: project/plugins.sbt (hunk 1 above)
Comment (huafengw) [inline project/plugins.sbt]: There is a new setting called packExclude that I need.

**Comment by huafengw** [comment]
This pull request just split the project core, next step is to split the lib folder.

File: project/plugins.sbt (hunk 1 above)
Comment (huafengw) [inline project/plugins.sbt]: I test the local mode, I'll test this in a cluster.

**Comment by kkasravi** [comment]
This will likely impact scalajs integration, we'll rebase the scalajs branch.

**Pull request merged**